ast/ddl.go

```diff
@@ -70,11 +70,13 @@ type CreateDatabaseStmt struct {
 }
 
 // Accept implements Node Accept interface.
-func (cd *CreateDatabaseStmt) Accept(v Visitor) (Node, bool) {
-	if skipChildren, ok := v.Enter(cd); skipChildren {
-		return cd, ok
+func (nod *CreateDatabaseStmt) Accept(v Visitor) (Node, bool) {
+	newNod, skipChildren := v.Enter(nod)
+	if skipChildren {
+		return v.Leave(newNod)
 	}
-	return v.Leave(cd)
+	nod = newNod.(*CreateDatabaseStmt)
```

Comment:
The returned type should be the same for Enter and Leave function. I'll add comments on Visitor methods.